feat: 3508 Added RenewalForecastUpdated and RenewalLikelihoodUpdated events to Timeline

Kasia9090 opened the pull request:
## Proposed changes

1. This commit includes new components and updates for handling 'RenewalForecastUpdated' and 'RenewalLikelihoodUpdated' events in the application timeline, enriching user interaction tracking. The GraphQL schemas and types have also been updated to accommodate the changes incorporated. This contributes to enhancing the functionality of the Timeline
2. Updated Timeline types and fix showing date condition for Meeting stub

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context

Kasia9090 committed:
feat 3508: Added RenewalForecastUpdated and RenewalLikelihoodUpdated events to Timeline

This commit includes new components and updates for handling 'RenewalForecastUpdated' and 'RenewalLikelihoodUpdated' events in the application timeline, enriching user interaction tracking. The GraphQL schemas and types have also been updated to accommodate the changes incorporated. This contributes to enhancing the functionality of the Timeline

Kasia9090 committed:
Refactor: Update Timeline types and fix showing date condition for Meeting stub

acalinica commented on packages/apps/spaces/app/organizations/[id]/components/Timeline/events/email/EmailStub.tsx:
```diff
@@ -5,12 +5,15 @@ import { Text } from '@ui/typography/Text';
+import { InteractionEventWithDate } from '../../types';
 
-export const EmailStub: FC<{ email: InteractionEvent }> = ({ email }) => {
+export const EmailStub: FC<{ email: InteractionEventWithDate }> = ({
```
Can we not use FC if we're not doing anything with the children? I've seen this in a lot of places and it's not helpful at all. If we are to be intentional about props containing children, we can use React.PropsWIthChildren otherwise typing only the props we use is just fine.

Kasia9090 replied:
In the new react version (which we use) FC does not include optional children props so it does not geopardise type safety
Using the React.FC<Props> approach provides a clear indication that a function is a functional React component. It also opens up the possibility for utilizing static properties such as defaultProps, propTypes, and contextTypes, which can further enhance component with default properties, enforcing property type checking, or using context within it.

Both approaches to type components are valid and it often comes down to personal preference, in my opinion if we want to enforce single standard I would propose to use FC for reasons mentioned above

acalinica replied:
propTypes is useless in context of typescript, defaultProps maybe can add some value although it's easier to just do `{ someProps = 'default' }` and adding default to properties is always better to be done from a data mapping layer as for contextTypes I don't know what to say... seems very abstract and low level and never ever used it. Nonetheless, my opinion about this is that we optimise ahead of time for something that didn't yet occurred. We already now that a react component must start with an `UpperCase` letter in the same manner we know a hook must start with `use`. A while ago the whole react community [advised](https://github.com/facebook/create-react-app/pull/8177) against React.FC and everybody started moving away from it. For me it was making sense when PropsWithChildren was not yet added to react typing and you always had to manually type the children and even then, people would make components with FC even though the children prop was never used. I've been though this once with FC and I really don't want to start refactoring away from it because it becomes infamous again.

Kasia9090 replied:
In the article where community advised to move out from FC was because in previous versions FC included children as optional props which was reducing type safety - this does not holds today as in new version FC does not include optional children anymore.

acalinica commented on packages/apps/spaces/app/organizations/[id]/components/Timeline/preview/TimelineEventsPreviewContext/TimelineEventPreviewContext.tsx:
```diff
@@ -1,33 +1,30 @@
+    RefObject,
+    useContext,
+    createContext,
+    PropsWithChildren,
```
This extra 2 spaces added for all those imports look odd. Are they formatted ok?

acalinica commented on packages/apps/spaces/app/organizations/[id]/components/Timeline/preview/TimelineEventsPreviewContext/TimelineEventPreviewContext.tsx:
```diff
@@ -1,33 +1,30 @@
 interface TimelineEventPreviewContextContextMethods {
-  containerRef: RefObject<HTMLDivElement> | null;
-  openModal: (content: Event) => void;
+  TimelineEventPreviewContextContainerRef: RefObject<HTMLDivElement> | null;
```
why do we need this long name for this ref?

acalinica commented on packages/apps/spaces/app/organizations/[id]/components/Timeline/events/action/RenewalForecastUpdatedAction.tsx:
```diff
@@ -0,0 +1,64 @@
+    metadata = data?.metadata && JSON.parse(data.metadata);
+  } catch (error) {
+    console.error('Error parsing metadata', error);
+  }
```
I would encapsulate the above code in a iife or a plain function to keep it away from the component scope. Also I think we should not log errors to the console intentionally (I agree that we should track this in sentry though.).

acalinica commented on packages/apps/spaces/app/organizations/[id]/components/Timeline/preview/TimelineEventsPreviewContext/TimelineEventPreviewContext.tsx:
```diff
@@ -43,19 +40,19 @@ export const TimelineEventPreviewContextContextProvider = ({
-  const [modalContent, setModalContent] = useState<Event | null>(null);
-  const containerRef = useRef<HTMLDivElement>(null);
+  const [modalContent, setModalContent] = useState<TimelineEvent | null>(null);
+  const TimelineEventPreviewContextContainerRef = useRef<HTMLDivElement>(null);
```
About this ref, last time I renamed it I have not seen it used anywhere. What's the purpose of it?

Kasia9090 committed:
Refactor code indentation in TimelineEventPreviewContext.tsx

Kasia9090 committed:
Remove unused code and debug logging

Removed 'console.error' logging from RenewalLikelihoodUpdatedAction.tsx and 'TimelineEventPreviewContextContainerRef' from TimelineEventPreviewContext.tsx.

Kasia9090 committed:
Update error handling in RenewalForecastUpdateAction component

Extracted JSON parsing of metadata into a separate 'getMetadata' function and replaced 'console.error' with Sentry's 'captureException' in RenewalForecastUpdatedAction.tsx. This change is aimed to improve error tracking and make exception handling more robust and centralized.
